### PR TITLE
Prepare release of GIMP 2.10.32.

### PR DESCRIPTION
**Pull request by Jehan**
- Bump babl and GEGL.
- Try to build the HEAD of the `gimp-2-10` branch.

@hfiguiere @HarryMichal We will normally release GIMP 2.10.32 on Sunday so I prepare a branch to test it all works fine. If there is anything to update or fix which you'd like to get in, in the same time, feel free to push commits to the branch `wip/release/2-10-32`.

### CODE REVIEW

File: org.gimp.GIMP.json [hunk 1]
```diff
@@ -716,8 +716,7 @@
                 {
                     "type": "git",
                     "url": "https://gitlab.gnome.org/GNOME/gimp.git",
-                    "tag": "GIMP_2_10_30",
-                    "commit": "cee406b593fb561efebc3664bdc0790fa45b9c47"
+                    "branch": "gimp-2-10"
```
Comment (HarryMichal) [inline org.gimp.GIMP.json:719]: Any reason for the switch from `tag`+`commit` to `branch`?

Comment (Jehan) [reply org.gimp.GIMP.json:719]: It's only temporary since we don't have a `GIMP_2_10_32` tag yet but we want to test with latest code **before** GIMP 2.10.32 is actually released (to detect problems early). So this is what I usually do once I start testing for a release: the first commit switch to `branch` to test the `HEAD` of the branch. Then the last commit switch back to `tag` once we have a tag published.

Comment (HarryMichal) [reply org.gimp.GIMP.json:719]: Ah, that makes sense. Thanks for the explanation!

**Comment by HarryMichal** [comment]
During tomorrow I'll try to wire up the dependency checker, so I can bump the versions of modules before the release. More work for less work \o/.

**Comment by novomesk** [comment]
Can you backport following part to the json file so that we have libjxl also in 2.10.32?

https://github.com/flathub/org.gimp.GIMP/blob/0d320aa793100737aa4f94ba46c4983f926a4aea/org.gimp.GIMP.json#L618-L643

**Comment by Jehan** [comment]
The aarch64 build failure is weird:

```
mv -f .deps/pangocairo_la-pangocairomodule.Tpo .deps/pangocairo_la-pangocairomodule.Plo
FB: host_command_exited_cb 462929 512
]2;flatpak-builder: Cleanup pygtkError: module pygtk: Child process exited with code 2
FB: Unmounting read-only fs: fusermount -uz /srv/buildbot/worker/build-aarch64/build/.flatpak-builder/rofiles/rofiles-x0kFBk
command timed out: 3600 seconds without output running [b'dbus-run-session', b'flatpak', b'run', b'org.flatpak.Builder', b'-v', b'--force-clean', b'--sandbox', b'--delete-build-dirs', b'--user', b'--install-deps-from=flathub', b'--install-deps-from=flathub-beta', b'--arch', b'aarch64', b'--mirror-screenshots-url=https://dl.flathub.org/repo/screenshots', b'--repo', b'repo', b'--extra-sources=/srv/buildbot/worker/build-aarch64/../downloads', b'--default-branch', b'test', b'--subject', b'Prepare release of GIMP 2.10.32. (807ed30b)', b'--remove-tag=upstream-maintained', b'builddir', b'org.gimp.GIMP.json'], attempting to kill
program finished with exit code 1
elapsedTime=5797.895253
```

Looking at previous lines, it's not clear at all what failed (if anything), or maybe if the build simply kind of locked and timed-out (cf. the "3600 seconds without output" message). In the later case, I wonder if it might just be a random failure as what is reported [here](https://github.com/flathub/flathub/issues/2894).
Otherwise, if anyone has the answer, feel free to fix! ;-)

The x86_64 build succeeded but just the release date being in the future, it fails at metadata validation. This AppStream data validation is sometimes so annoying.

**Comment by HarryMichal** [comment]
There seems to be this (starting at line 164172 of the aarch64 build log):

```
...
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/app/include/python2.7 -I/app/include/python2.7 -I/app/include/pygtk-2.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -I./gtk -I/usr/include/atk-1.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -Wall -fno-strict-aliasing -std=c9x -MT atk_la-atkmodule.lo -MD -MP -MF .deps/atk_la-atkmodule.Tpo -c atkmodule.c  -fPIC -DPIC -o .libs/atk_la-atkmodule.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/app/include/python2.7 -I/app/include/python2.7 -I/app/include/pygtk-2.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -I./gtk -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -pthread -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -Wall -fno-strict-aliasing -std=c9x -MT pango_la-pangomodule.lo -MD -MP -MF .deps/pango_la-pangomodule.Tpo -c pangomodule.c  -fPIC -DPIC -o .libs/pango_la-pangomodule.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/app/include/python2.7 -I/app/include/python2.7 -I/app/include/pygtk-2.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -pthread -I./gtk -I/app/include/pycairo -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -pthread -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/app/include/pycairo -pthread -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -Wall -fno-strict-aliasing -std=c9x -MT pangocairo_la-pangocairomodule.lo -MD -MP -MF .deps/pangocairo_la-pangocairomodule.Tpo -c pangocairomodule.c  -fPIC -DPIC -o .libs/pangocairo_la-pangocairomodule.o
Traceback (most recent call last):
  File "/app/share/pygobject/2.0/codegen/codegen.py", line 11, in <module>
    import defsparser
  File "/app/share/pygobject/2.0/codegen/defsparser.py", line 3, in <module>
    import scmexpr
EOFError: EOF read where object expected
make: *** [Makefile:1151: pango.c] Error 1
make: *** Waiting for unfinished jobs....
...
```

This happened in a module that has not been changed by this PR, so I'm tempted to blame it on the build runner and just try to rebuild. But that seems to be too easy.

**Comment by hfiguiere** [comment]
I would just retrigger the build, after fixing the appadata.

**Comment by HarryMichal** [comment]
Would be nice, if GIMP builds before I push updates to the modules.

**Comment by Jehan** [comment]
> I would just retrigger the build, after fixing the appadata.

The appdata is not broken, just in the future.
I guess I could just set it to the wrong date (e.g. today), but it bothers me to push a knowingly wrong commit in the main repo only to undo it in a few days. Maybe we could do the other way around. Patch the appdata here, allowing us to test, and not forget to remove the commit before actual release.

**Comment by hfiguiere** [comment]
I would just patch here. Either a patch or some sed magic.

**Comment by hfiguiere** [comment]
btw I suspect the aarch64 build was an infra problem.

**Comment by HarryMichal** [comment]
Local build still hasn't finished but just to make sure, should I push the module updates (together with the dependency checker work) before the new test build or after it?

**Comment by Jehan** [comment]
@HarryMichal Doesn't really matter. You can push already and it won't cancel the previous build either. It will just create a new build.

So yeah, push. No prob.

**Comment by HarryMichal** [comment]
I briefly tried the build with the updated modules and at first glance everything seemed to be working fine.

**Comment by Jehan** [comment]
bot, build

**Comment by Jehan** [comment]
@HarryMichal From a local flatpak build (since the test builds choked on appdata)?

I restarted a test build anyway.

**Comment by HarryMichal** [comment]
> @HarryMichal From a local flatpak build (since the test builds choked on appdata)?

Yes, local build.

**Comment by Jehan** [comment]
bot, build

**Comment by Jehan** [comment]
Freedesktop's gitlab is down. Cairo can't be downloaded. Let's just merge tomorrow (hoping it's back up). :-)

**Comment by HarryMichal** [comment]
bot, build

**Comment by HarryMichal** [comment]
Yikes, I thought the service was up again but I tried out a wrong URL. It's still down.

**Comment by hfiguiere** [comment]
I was reading this in another issue:
https://github.com/flathub/org.rncbc.qpwgraph/pull/8#issuecomment-1153739783

tl;dr might not be fixed until next month.

**Comment by Jehan** [comment]
bot, build

**Comment by hfiguiere** [comment]
> I was reading this in another issue: [flathub/org.rncbc.qpwgraph#8 (comment)](https://github.com/flathub/org.rncbc.qpwgraph/pull/8#issuecomment-1153739783)
> 
> tl;dr might not be fixed until next month.

Update: it's actually a disk problem (no data loss) that make the cluster to be impaired. It's being worked on.
